back/src/Guide/Db/Get.hs

```diff
@@ -0,0 +1,87 @@
+import Guide.Markdown (toMarkdownInline, toMarkdownBlock, toMarkdownTree)
+import Guide.Utils (Uid(..))
+
+
+-- |
+getTest :: IO ()
+getTest = do
+  conn <- connect
+  trait <- run' (getItemByItemIdMaybe "items1234567") conn
+  print trait
+
+-- | Methods to get information from database
+
+-- | Get trait by id, either it deleted or not.
+getTraitByTraitIdMaybe :: Uid Trait -> Session (Maybe Trait)
+getTraitByTraitIdMaybe (Uid uidTrait) = do
+  let sql = toByteString [text|
+        SELECT uid, content FROM traits WHERE uid = '${uidTrait}';|]
+      encoder = HE.noParams
+      decoder = HD.rowMaybe (Trait
+        <$> (Uid <$> (HD.column . HD.nonNullable) HD.text)
+        <*> (toMarkdownInline <$> (HD.column . HD.nonNullable) HD.text))
+  HS.statement () (Statement sql encoder decoder False)
+
+-- | Get traits list by item id, deleted or not and trat taip.
+getTraitByItemIdMaybe :: Bool -> Text -> Uid Item -> Session [Trait]
+getTraitByItemIdMaybe isDeleted traitType (Uid uidItem) = do
+  -- TODO: ProCon datatype to pass here not a text
+  let deletedT = toText $ show isDeleted
+  let sql = toByteString [text|
+        SELECT uid, content FROM traits
+          WHERE item_uid = '${uidItem}' AND
+                deleted = '${deletedT}' AND
+                type_ = '${traitType}'
+        ;|]
+      encoder = HE.noParams
+      decoder = HD.rowList (Trait
+        <$> (Uid <$> (HD.column . HD.nonNullable) HD.text)
+        <*> (toMarkdownInline <$> (HD.column . HD.nonNullable) HD.text))
+  HS.statement () (Statement sql encoder decoder False)
+
+-- | Get item by id, either it deleted or not.
+getItemByItemIdMaybe :: Uid Item -> Session (Maybe Item)
+getItemByItemIdMaybe uid@(Uid uidItem) = do
+  pro <- getTraitByItemIdMaybe False "pro" uid
+  proDeleted <- getTraitByItemIdMaybe True "pro" uid
+  con <- getTraitByItemIdMaybe False "con" uid
+  conDeleted <- getTraitByItemIdMaybe True "con" uid
+  let sql = toByteString [text|
+        SELECT * FROM items WHERE uid = '${uidItem}';|]
+      encoder = HE.noParams
+      decoder = HD.rowMaybe (Item
+        <$> (Uid <$> (HD.column . HD.nonNullable) HD.text)                             -- uid
+        <*> ((HD.column . HD.nonNullable) HD.text)                                     -- name
+        <*> (localTimeToUTC utc <$> (HD.column . HD.nonNullable) HD.timestamp)         -- localtime
+        <*> ((HD.column . HD.nullable) HD.text)                                        -- group_
+        <*> ((HD.column . HD.nullable) HD.text)                                        -- link
+        <*> ((HD.column . HD.nullable) HD.text)                                        -- hackage
+        <*> (toMarkdownBlock <$> (HD.column . HD.nonNullable) HD.text)                 -- summary
+        <*> (toMarkdownBlock <$> (HD.column . HD.nonNullable) HD.text)                 -- ecosystem
+        <*> (toMarkdownTree "whatisidprefix" <$> (HD.column . HD.nonNullable) HD.text) -- notes. What a prefix?
```

Comment:
@neongreen What is it prefix about?

Reply:
https://github.com/aelve/guide/blob/develop/back/src/Guide/State.hs#L388-L389

The prefix specifies how `id`s for headers should look in rendered HTML. Let's say two items' notes both look like this:

```markdown
# Problems
...

# Solutions
...
```

If we generate `<h1 id="problems">` for both of those, linking to a particular part of a particular note with `guide.aelve.com/haskell/someid#problems` becomes impossible. The solution is to have a different prefix for all headers in notes. Then links will look like this: `guide.aelve.com/haskell/someid#item-notes-someid-problems`.